### ISSUE TITLE
Allow a project to have multiple extensions of type NamedDomainObjectContainer

### ISSUE THREAD

**Issue by adammurdoch** [gradle/kotlin-dsl#905]
When there are multiple extensions of type `NamedDomainObjectContainer` defined for a project, then the generated accessors conflict and script compilation fails.

This is a problem when more than one plugin uses `NamedDomainObjectContainer` as an extension.

### Expected Behavior

Multiple plugins can use an extension of type `NamedDomainObjectContainer`.

### Current Behavior

Script compilation fails when multiple extensions of type `NamedDomainObjectContainer` defined for a project.

### Context

This is blocking https://github.com/gradle/gradle-native/issues/681

### Steps to Reproduce (for bugs)

Given this In a plugin:

```
project.getExtensions().add("strings", project.container(String.class));
project.getExtensions().add("longs", project.container(Long.class));
```

Then, script compilation fails with:

```
e: /Users/adam/.gradle/caches/4.9-20180604044642+0000/gradle-kotlin-dsl-accessors/4nhiy17kxkystvtpaifdugx8f/src/org/gradle/kotlin/dsl/accessors.kt:56:110: Platform declaration clash: The following declarations have the same JVM signature (getExt(Lorg/gradle/api/NamedDomainObjectContainer;)Lorg/gradle/api/plugins/ExtraPropertiesExtension;):
    fun NamedDomainObjectContainer<Long>.<get-ext>(): ExtraPropertiesExtension defined in org.gradle.kotlin.dsl in file accessors.kt
    fun NamedDomainObjectContainer<String>.<get-ext>(): ExtraPropertiesExtension defined in org.gradle.kotlin.dsl in file accessors.kt
```

**Comment by bamboo** [comment]
Closing as delivered.

**Comment by hzsweers** [comment]
What does delivered mean in this context?

**Comment by bamboo** [comment]
It means the feature made into `develop` and it will be available in the next release.

**Comment by hzsweers** [comment]
:+1:

**Comment by p-schneider** [comment]
The fix included in [kotlin-dsl version 0.18.4](https://github.com/gradle/kotlin-dsl/releases/tag/v0.18.4) made it into the stable [gradle version 4.9](https://github.com/gradle/gradle/releases/tag/v4.9.0) release today.
